### PR TITLE
Add support for powering the bus with the pin.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/one_wire.toit [hunk 1]
```diff
@@ -31,13 +31,22 @@ class Bus:
...
+  /**
+  The protocol.
+  Use the $protocol_ accessor to use the protocol.
+  */
+  protocol__/Protocol? := ?
+
+  protocol_ --power/bool [block] -> none:
+    if not protocol__: throw "BUS CLOSED"
+    protocol__.set_power power
+    block.call protocol__
```
Comment (erikcorry) [inline src/one_wire.toit:43]: I don't understand the benefit of the block.  Nothing is done after the block.  Why not just return the protocol_ ?

File: src/one_wire.toit [hunk 2]
```diff
@@ -31,13 +31,22 @@ class Bus:
...
 
   /**
   Constructs a 1-wire bus using the given $protocol.
   */
   constructor.protocol protocol:
```
Comment (erikcorry) [inline src/one_wire.toit:48]: Why is the protocol argument not typed?

**Review by erikcorry — APPROVED** [review]

File: src/one_wire.toit (hunk 1 above)
Comment (floitsch) [reply src/one_wire.toit]: Agreed. It's easier to just return the protocol.
done.

File: src/one_wire.toit (hunk 2 above)
Comment (floitsch) [reply src/one_wire.toit]: Initially it was using a `.protocol_` (at which point it got the type from the field).
Added the type.
done.

**Pull request merged**